ts/navigation/MainNavigator.tsx

```diff
@@ -42,6 +36,18 @@ const ROUTE_ICON: RouteIconMap = {
   MAIN_PROFILE: "io-profilo"
 };
 
+const getLabel = (routeName: string): string => {
+  const fallbackLabel = "unknown"; // fallback label
+  const label = ROUTE_LABEL[routeName as Routes]; // routeName is defined as string, but has values within Routes
```

Comment:
the problem here is that somewhere we're "losing" the type of the route name that should be of type `key in Routes` - the problem with the cast is basically that you're saying to the TS compiler: _don't bother me, I know what I'm doing, I'm sure that `routeName` is a `key in Routes`_. So I guess my question is, are we really sure? If so, add a comment like "we're casting because we're losing the type of routeName but we're sure it's a `key in Routes` because..."

Reply:
`routeName` is typed, as already mentioned, as a `string`, so you'll never have the mathematical certainty that that value will be outside of the values in `Routes` (hence the fallback option).

Despite that, I'll try to argument why I reasonably assumed that `routeName` will always be within `Routes`. 

`TabNavigator` is instantiated in `MainNavigator`:
```
const navigation = TabNavigator(
  {
    [ROUTES.MAIN_MESSAGES]: { screen: ... },
    [ROUTES.MAIN_PROFILE]: { screen: ... },
  }, ...
);
```

`TabNavigator`'s definition passes the first argument as`routeConfigs` to a `TabRouter`.

`TabRouter`'s method `getStateForAction()` defines 
```
  const order = config.order || Object.keys(routeConfigs);
```
and uses `order` throughout it to populate the state which is then used in `getLabel` and `getIcon` (through `nav.state.routName`)). Thus, since `config.order` is not specified (it is part of the config object passed as the second argument to `TabNavigator`), `order` is the list of keys from `routeConfig`. 

This applies for react-navigator 1.5.12, following versions are deprecating the Navigator ([#157348932](https://www.pivotaltracker.com/story/show/157348932)).

Is this the kind of reasoning you were looking for?

Reply:
that would be perfect, you can even link to this comment if you want, with a comment like `See this comment on why this cast is safe: [link]`